plugins/CoreHome/DataTableRowAction/RowEvolution.php

```diff
@@ -111,8 +112,12 @@ public function __construct($idSite, $date, $graphType = 'graphEvolution')
 
         if ($this->period != 'range') {
             // handle day, week, month and year: display last X periods
+            //handle cache if exist
+            $cache = ViewDataTableManager::getViewDataTableParameters(Piwik::getCurrentUserLogin(),
+              'CoreHome.getRowEvolutionGraph');
+            $lastDay = !empty($cache) ? $cache['evolution_' . $this->period . '_last_n'] : null;
```

Comment:
```suggestion
            $lastDay = (isset($cache['evolution_' . $this->period . '_last_n']) ? $cache['evolution_' . $this->period . '_last_n'] : null);
```
The cache object might be non-empty but still not contain `evolution_xxx_last_n` key